mws/resource_log_delivery.go

```diff
@@ -0,0 +1,131 @@
+package mws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databrickslabs/databricks-terraform/common"
+	"github.com/databrickslabs/databricks-terraform/internal"
+	"github.com/databrickslabs/databricks-terraform/internal/util"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+// LogDelivery wrapper
+type LogDelivery struct {
+	LogDeliveryConfiguration LogDeliveryConfiguration `json:"log_delivery_configuration"`
+}
+
+// LogDeliveryList lists log delivery configurations
+type LogDeliveryList struct {
+	LogDeliveryConfigurations []LogDeliveryConfiguration `json:"log_delivery_configurations"`
+}
+
+// LogDeliveryConfiguration describes log delivery
+type LogDeliveryConfiguration struct {
+	AccountID              string   `json:"account_id"`
+	ConfigID               string   `json:"config_id,omitempty" tf:"computed"`
+	CredentialsID          string   `json:"credentials_id"`
+	StorageConfigurationID string   `json:"storage_configuration_id"`
+	WorkspaceIdsFilter     []string `jsong:"workspace_ids_filter,omitempty"`
+	ConfigName             string   `json:"config_name,omitempty"`
+	Status                 string   `json:"status,omitempty"`
+	LogType                string   `json:"log_type"`
+	OutputFormat           string   `json:"output_format"`
+	DeliveryPathPrefix     string   `json:"delivery_path_prefix,omitempty"`
+	DeliveryStartTime      string   `json:"delivery_start_time,omitempty"`
+}
+
+// LogDeliveryAPI ...
+type LogDeliveryAPI struct {
+	client *common.DatabricksClient
+}
+
+// NewLogDeliveryAPI ...
+func NewLogDeliveryAPI(m interface{}) LogDeliveryAPI {
+	return LogDeliveryAPI{client: m.(*common.DatabricksClient)}
+}
+
+// Read reads log delivery configuration
+func (a LogDeliveryAPI) Read(accountID, configID string) (LogDeliveryConfiguration, error) {
+	var ld LogDelivery
+	err := a.client.Get(fmt.Sprintf("/accounts/%s/log-delivery/%s", accountID, configID), nil, &ld)
+	return ld.LogDeliveryConfiguration, err
+}
+
+// Create new log delivery configuration
+func (a LogDeliveryAPI) Create(ldc LogDeliveryConfiguration) (string, error) {
+	var ld LogDelivery
+	err := a.client.Post(fmt.Sprintf("/accounts/%s/log-delivery", ldc.AccountID), LogDelivery{
+		LogDeliveryConfiguration: ldc,
+	}, &ld)
+	// todo: verify with empty response - structs should have empty default strings
+	return ld.LogDeliveryConfiguration.ConfigID, err
```

Comment:
why are we just returning configId here and then making the read call again in CreateContext?

Reply:
this concept is called "resource drift detection" - https://www.terraform.io/docs/extend/best-practices/detecting-drift.html - and is followed across all API integrations in provider